ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -166,7 +164,45 @@ def internal_delete_uls(self, name):
            name (str): The Underlying Storage name to be deleted
 
         """
-        assert self.exec_uls_deletion(name)
+        def _exec_uls_deletion(self, name):
+            """
+            Try to delete Underlying Storage by name if exists
+
+            Args:
+                name (str): the Underlying Storage name
+
+            Returns:
+                bool: True if deleted successfully
+
+            """
+            if self.verify_uls_exists(name):
+                try:
+                    if 'aws' in name:
+                        self.client.meta.client.delete_bucket_policy(
```

Comment:
just for a better understanding, don't we have to delete the bucket policy on other cloud providers as well?

Reply:
At the moment, the only other provider that uses S3 is IBM COS.
However, whenever I tried to run this line with an IBM bucket, it failed with an error saying it's not supported - maybe because IBM handles the deletion of a non-existent bucket policy in a different way than AWS (that simply ignores it and executes without a problem).
At the same time, IBM *does* support bucket policies from what I've read, so I'm marking this as todo.

Reply:
Ok, thanks for the explanation. Adding @nimrod-becker to provide his input if he has any